src/ServerlessOffline.js

```diff
@@ -369,10 +369,14 @@ export default class ServerlessOffline {
             }
 
             httpEvent.http.isHttpApi = true
-            httpEvent.http.payload =
-              service.provider.httpApi && service.provider.httpApi.payload
-                ? service.provider.httpApi.payload
-                : '2.0'
+            if (functionDefinition.httpApi && functionDefinition.httpApi.payload) {
+              httpEvent.http.payload = functionDefinition.httpApi.payload;
```

Comment:
I believe in that case eventual setting on `httpEvent` should take precedence

Reply:
`httpEvent`? Do you mean the provider setting? If so, isn't that counterintuitive?

Reply:
Sorry, ignore my comment. There's no setting on event directly (it obviously wouldn't make sense :)